addons/website_sale_autocomplete/models/autocompletion_provider.py

```diff
@@ -0,0 +1,145 @@
+# -*- encoding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import requests
+
+from odoo import models, fields
+from odoo.tools import html2plaintext
+
+FIELDS_MAPPING = {
+    'country': ['country'],
+    'street_number': ['number'],
+    'locality': ['city'],  # If locality exists, use it instead of the more general administrative area
+    'route': ['street'],
+    'postal_code': ['zip'],
+    'administrative_area_level_1': ['state', 'city'],
+    'administrative_area_level_2': ['state', 'country']
+}
+
+# If a google fields may correspond to multiple standard fields, the first occurrence in the list will overwrite following entries.
+FIELDS_PRIORITY = ['country', 'street_number', 'neighborhood', 'locality', 'route', 'postal_code',
+                   'administrative_area_level_1', 'administrative_area_level_2']
+GOOGLE_PLACES_ENDPOINT = 'https://maps.googleapis.com/maps/api/place'
+
+
+class AutoCompletionProvider(models.Model):
+    """
+    As the Google API is very inconsistent with it's answers, this class also serves as a decoder for translating Google Data into a standard format.
+    """
+    _name = 'autocompletion.provider'
+    _description = 'Suggests results from a search string using external APIs/Tools'
+    _order = 'name,id'
+
+    name = fields.Char(required=True)
+    provider = fields.Selection([
+        ('google_places', 'Google Places')
+    ], required=True, default='google_places')
+
+    search_type = fields.Selection([
+        ('none', 'None'),
+        ('address', 'Address')
+    ], required=True, default='none')
+
+    def _get_key(self):
+        return self.env['website'].browse(self.env.context['website_id']).google_places_api_key
```

Comment:
We should use the `get_current_website` method instead here.
```suggestion
        return self.env['website'].get_current_website().google_places_api_key
```